Update Orchestrator to call CustomQuestionAnswering

Arsh-Kashyap opened the pull request:
#3737

## Proposed Changes
Updated C# and NodeJS 14.nlp-with-orchestrator sample to call Custom Question Answering.



## Testing
<!-- If you are adding a new feature to a library, you must include tests for your new code. -->

sahithikkss reviewed (commented):
correct typo `01.dispatch-bot` to `14.nlp-with-orchestrator` in C# sample readme.md

akhilayesantarao99 commented on samples/javascript_nodejs/14.nlp-with-orchestrator/bots/dispatchBot.js:
```diff
@@ -78,19 +79,19 @@ class DispatchBot extends ActivityHandler {
-            console.log(`Dispatch unrecognized intent: ${ intent }.`);
-            await context.sendActivity(`Dispatch unrecognized intent: ${ intent }.`);
-            break;
+            case 'HomeAutomation':
```
undo whitespace changes

Arsh-Kashyap replied:
Switch-case didn't have the correct spacing, so eslint extension took care of it.

Arsh-Kashyap commented on samples/csharp_dotnetcore/14.nlp-with-orchestrator/Bots/DispatchBot.cs:
```diff
@@ -58,7 +58,7 @@ private async Task DispatchToTopIntentAsync(ITurnContext<IMessageActivity> turnC
                     await ProcessWeatherAsync(turnContext, cancellationToken);
                     break;
-                case "QnAMaker":
+                case "CustomQA":
```
We have been using CustomQABot and CustomQABotAllFeatures in bot samples, and this would be in line with that. CQA might be too short.

Arsh-Kashyap commented on samples/csharp_dotnetcore/14.nlp-with-orchestrator/BotServices.cs:
```diff
@@ -20,18 +22,19 @@ public BotServices(IConfiguration configuration, OrchestratorRecognizer dispatch
-        public QnAMaker SampleQnA { get; private set; }
-        
+
+        public IQnAMakerClient SampleQnA { get; private set; }
```
Changed it to CustomQuestionAnswering.

gabog reviewed (changes requested):
Can we create a proper issue for this PR rather than using #minor (it is definitely not minor and has impact on docs as well)

We need to discuss what our recommendation is and how would existing customers will have access to the older version of this sample if they need to.

sahithikkss commented:
> @gabog , would it be possible to have a commit snapshot of current main as 4.15 samples? Can we redirect customers looking for QnA maker version of sample to that?
Also, documentation changes are also on our to-do list and is being tracked by our PM @shshubhe.

Arsh-Kashyap commented:
@gabog , To solve the documentation problem and confusion on which sample to point for orchestrator with qnamaker, we discussed internally to have a separate sample for custom question answering. I have added a new sample each in C# and NodeJS,  and removed the deprecation notice. Deprecation notice may be added to the old samples and we can point to these new samples for orchestrator with CQA.
